Expose optional native error details for auth failures

KevinvdBurg opened the pull request:
## Description

Surfaces underlying native authorization errors on `error.nativeError` while keeping `error.message` unchanged (e.g. `Network error`).

Apps can keep showing `error.message` to users and use `error.nativeError` when they want the lower-level cause for debugging or custom UI.

This revives the original Android improvement, updates it for the current package layout, and extends the same behavior to iOS.

Related: #861, #799.

## Verification

- Verified Android locally with the demo app by forcing the IdentityServer token endpoint to use `https://expired.badssl.com`.
- Confirmed `error.message` stays `Network error` on both platforms.
- Confirmed the demo can consume `error.nativeError` and surface the certificate/cause detail (see screenshots below).
- Built and launched the iOS demo app, then verified the same broken token endpoint flow on iOS.

| Android | iOS |
| :-: | :-: |
| <img width="642" height="1172" alt="image" src="https://github.com/user-attachments/assets/ef7432b9-46c6-4060-aff2-0d13e2c98bbe" /> | <img width="568" height="1084" alt="image" src="https://github.com/user-attachments/assets/f10b3a75-a0dc-4c2f-a860-696f19c7196f" /> |

KevinvdBurg committed:
Added a more descriptive error message for AuthorizationExceptions

KevinvdBurg committed:
Added extra space before - Cause

robwalkerco commented:
Thanks for your PR @KevinvdBurg 

I noticed that you only modified the error description for Android. Is iOS already providing this additional error information? If not, would you be able to update your PR to keep iOS and Android aligned?

KevinvdBurg commented:
Hi @robwalkerco,

I'm trying to open the iOS project, but I'm getting the following error 🤔 
<img width="278" alt="image" src="https://github.com/FormidableLabs/react-native-app-auth/assets/4567644/ded80ad9-4ef4-420f-8c12-a7fa1e55dc8b">

But also, I'm unsure where the iOS variant `handleAuthorizationException` is the naming is not consistent between the platforms. If you can point me to the right function, I can add it. :)

bryceknz committed:
chore: sync pr 886 with main

bryceknz committed:
fix(android): include authorization exception causes

bryceknz committed:
fix(ios): include underlying authorization error causes

bryceknz committed:
chore: add changeset for clearer native errors

bryceknz commented:
Thanks for this contribution, @KevinvdBurg!

I was able to verify the Android behavior locally, and the added cause detail makes the error much more useful for debugging.

I’m going to continue the work from here by bringing the branch up to date with the current repo structure and extending the same underlying-error detail to iOS as well.

I'll make sure to give you git credit when I merge it!

bryceknz committed:
fix: expose native authorization errors separately

Keep user-facing auth error messages unchanged while surfacing underlying native diagnostics through error.nativeError.

bryceknz committed:
chore: update native error changeset

Document the separate nativeError debug field without suggesting native details belong in user-facing messages.

bryceknz committed:
fix: preserve native authorization exception context

Pass Android nativeError metadata without dropping the original AuthorizationException, and align the changeset frontmatter with repo style.

bryceknz committed:
chore: mark native error changeset minor
